topology/graph/graph.go

```diff
@@ -904,6 +910,10 @@ func (t *MetadataTransaction) Commit() {
 		}
 	}
 
+	for _, k := range t.removes {
+		common.DelField(e.metadata, k)
```

Comment:
It would be nice if `DelField` could also return a boolean that would be true if an element was really deleted and false otherwise. That way, we would not generate an event if the metadata were not modified.

Reply:
even an inbuilt delete function will not return any boolean value. we need to implement the logic. so I am thinking of a separate PR. is that fine for you?

Reply:
Now that DelField returns a boolean, we can do :
`updated = common.DelField(e.metadata, k) || updated`